research/README.md

```diff
@@ -0,0 +1,325 @@
+
+<img width='600' src='img/002-add-file-desktop-3.png' />
+
+**Companion** adds extra sharing related links to the menu when the current tab is showing a file from IPFS. "Copy Canonical Address", copies a Nested URI `/ipfs/QmHash` address to the clipboard. "Copy Public Gateway URL", copies the full ipfs.io url for the file. Both indicate that it worked with a browser notification. There is no indication to the user that these extra features exist, they only see them by re-opening the ifps browser action menu.
+
+<img width='888' src='img/002-add-file-companion-3.png' />
+
+## Browsing files
+
+Discussion: https://github.com/ipfs-shipyard/pm-ipfs-gui/issues/9
+
+**WebUI** list files by CID. It's not clear what order they are sorted in. Links are provided to open each file in a browser, or explore IPLD (labelled as DAG) data for the node are provided, along with a "x - remove" link.
+
+**Desktop** lists files by filename, sorted with most recently added at the top. The time the file was added is provided as muted, secondary text. In the 0.3, you can't drill down into folders, though I think this feature is available in the latest dev version.
+
+**Companion** has no file browsing feature.
+
+## Connections / Peers
+
+Desktop and WebUI both have a page for showing info about all the other IPFS nodes you are currently connected to.
+
+| webui | desktop |
+|-------|---------|
+| ![](img/003-peers-webui-1.png) | ![](img/003-peers-desktop-1.png) |
+
+**WebUI** shows a peer count, geolocated pins on a globe and list of Peer Id / Multiaddr pairs for each connection. Clicking on a list item expands it to show similar node info as show on the Home page about your local node. The process is resource intensive and causes "slow script" warnings in firefox.
+
+**Desktop** Shows your location and peer count, along with a list of Peer Id / Location pairs. No more info is available about the peers. There is a search bar that allows you to search for peers by id or location.
+
+**Companion** shows just the number of connected peers as a menu bar stat, and again when you go to share a file.
+
+## DAG / IPLD Explorer
+
+Discussion: https://github.com/ipfs-shipyard/pm-ipfs-gui/issues/11
+
+**WebUI** has a DAG explorer, which might be due a rename to "IPLD Explorer". It takes an IPFS hash (also due a rename to CID) and lists out all the links or data that in the object the CID points to. In the case of a directory, you can see a list of the files it contains, and each filename is a link to it's IPLD object, allowing you drill down, navigating through the links. Where a hash points to data, the raw data is rendered in plain text.
+
+| Empty state | Explore a cid | Drill into a data node |
+|-|-|-|
+|![](img/004-ipld-explorer-webui-1.png) |![](img/004-ipld-explorer-webui-2.png) |![](img/004-ipld-explorer-webui-3.png)
+
+The UI/UX should be improved but the idea is interesting, and could become a vital tool for helping people to understand IPFS and IPLD.
+
+## Pinning
```

Comment:
For completeness, let's add short note that **Companion** supports recursive pinning of arbitrary resources by navigating to the resource root CID and clicking on "Pin" in Browser/Page Action.